src/ints/mouse_dos_driver.cpp

```diff
@@ -0,0 +1,1440 @@
+
+#define CURSOR_SIZE_X  16
+#define CURSOR_SIZE_Y  16
+#define CURSOR_SIZE_XY (CURSOR_SIZE_X * CURSOR_SIZE_Y)
+
+#define NUM_BUTTONS 3
+
+enum class MouseCursor : uint8_t { Software = 0, Hardware = 1, Text = 2 };
+
+// These values represent 'hardware' state, not driver state
+
+MouseButtons12S buttons = 0;
+float pos_x     = 0.0f;
+float pos_y     = 0.0f;
+int16_t wheel   = 0;
+uint8_t rate_hz = 0; // TODO: add proper reaction for 0 (disable driver)
+
+static struct { // DOS driver state
+
+    // Structure containing (only!) data which should be
+    // saved/restored during task switching
+
+    // DANGER, WILL ROBINSON!
+    ///
+    // This whole structure can be read or written from the guest side
+    // via virtual DOS driver, functions 0x15 / 0x16 / 0x17.
+    // Do not put here any array indices, pointers, or anything that
+    // can crash the emulator if filled-in incorrectly, or that can
+    // be used by malicious code to escape from emulation!
+
+    bool enabled    = false; // TODO: make use of this
+    bool cute_mouse = false;
+
+    uint16_t times_pressed[NUM_BUTTONS]   = {0};
+    uint16_t times_released[NUM_BUTTONS]  = {0};
+    uint16_t last_released_x[NUM_BUTTONS] = {0};
+    uint16_t last_released_y[NUM_BUTTONS] = {0};
+    uint16_t last_pressed_x[NUM_BUTTONS]  = {0};
+    uint16_t last_pressed_y[NUM_BUTTONS]  = {0};
+    uint16_t last_wheel_moved_x           = 0;
+    uint16_t last_wheel_moved_y           = 0;
+
+    float mickey_x = 0.0f;
+    float mickey_y = 0.0f;
+
+    float mickeys_per_px_x = 0.0f;
+    float mickeys_per_px_y = 0.0f;
+    float pxs_per_mickey_x = 0.0f;
+    float pxs_per_mickey_y = 0.0f;
+
+    uint16_t gran_x = 0; // granularity mask
+    uint16_t gran_y = 0;
+
+    int16_t update_region_x[2] = {0};
+    int16_t update_region_y[2] = {0};
+
+    uint16_t language = 0; // language for driver messages, unused
+    uint8_t mode      = 0;
+
+    // sensitivity
+    uint16_t senv_x_val       = 0;
+    uint16_t senv_y_val       = 0;
+    uint16_t dspeed_threshold = 0; // threshold, in mickeys/s  TODO: should
+                                   // affect mouse movement
+    float senv_x = 0.0f;
+    float senv_y = 0.0f;
+
+    // mouse position allowed range
+    int16_t minpos_x = 0;
+    int16_t maxpos_x = 0;
+    int16_t minpos_y = 0;
+    int16_t maxpos_y = 0;
+
+    // mouse cursor
+    uint8_t page                     = 0; // cursor display page number
+    bool inhibit_draw                = false;
+    uint16_t hidden                  = 0;
+    uint16_t oldhidden               = 0;
+    int16_t clipx                    = 0;
+    int16_t clipy                    = 0;
+    int16_t hot_x                    = 0; // cursor hot spot, horizontal
+    int16_t hot_y                    = 0; // cursor hot spot, vertical
+    bool background                  = false;
+    uint16_t backposx                = 0;
+    uint16_t backposy                = 0;
+    uint8_t backData[CURSOR_SIZE_XY] = {0};
```

Comment:
Should be `back_data`. But maybe spell `background` out for all these.

Reply:
Changed naming as suggested below.